### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by prakashiit**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+RUN mkdir /daksha
+
+# Set the working directory to /ui_automation_engine
+WORKDIR /daksha
+
+# Copy the current directory contents into the container at /ui_automation_engine
+ADD . /daksha/
+
+# Install any needed packages specified in requirements.txt
```
Comment (SriMouli) [inline Dockerfile:15]: Dockerfile is incomplete. move items from docker-compose to here

File: README.md [hunk 2]
```diff
@@ -1,2 +1,40 @@
...
+===============================================
+
+## Introduction
+Daksha is a django framework based dynamic test case execution API.You can write any variation of UI-based test cases in a yml file and execute them right away with this new tool in town. <br /> For example: Sign in to any of your social media account and sign out or sign in then perform a task and then sign out. <br /> The API request should comprise of the location of the yml file.To summarize in simple terms the working can be divided into two steps -
+* [Sending API Request](daksha_know-how/ApiRequest.md)
+* [Writing Test Case](daksha_know-how/CreateTest.md)
+
+## Building and Running
+
+### Using docker-compose
```
Comment (SriMouli) [inline README.md:11]: change to using docker

File: README.md [hunk 3]
```diff
@@ -1,2 +1,40 @@
...
+
+* Open an issue at: https://github.com/mykaarma/daksha/issues
+* Email us at: opensource@mykaarma.com
+
+## Contributions
+
+We welcome contributions to Daksha. Please see our [contribution guide](CONTRIBUTING.md) for more details.
+
+## License
+Copyright 2020 myKaarma
```
Comment (SriMouli) [inline README.md:34]: change to 2021

File: daksha/settings.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'gj%7t91&c$*q&+fkx-wuug^i$0b@@lk26vd4txdq=)%_319hyy'
```
Comment (SriMouli) [inline daksha/settings.py:39]: change to env

File: daksha/settings.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = [
```
Comment (SriMouli) [inline daksha/settings.py:44]: is env support present for this?

File: daksha/settings.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+]
+
+MIDDLEWARE = [
+    'django.middleware.security.SecurityMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    # 'django.middleware.csrf.CsrfViewMiddleware',
```
Comment (SriMouli) [inline daksha/settings.py:64]: remove commented

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+version: '3'
+
+services:
+  web:
+    build: .
+    command: bash -c "python manage.py runserver 0.0.0.0:8000"
```
Comment (SriMouli) [inline docker-compose.yml:6]: use wsgy or some webServer. runserver is not prod ready

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    container_name: daksha
+    restart: always
+    volumes:
+      - /var/log/kaarya/:/var/log/kaarya/
```
Comment (SriMouli) [inline docker-compose.yml:10]: remove this

File: engine/email_generator.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        message = "Please open this url to get your test Report : " + report_url + "\nTest folder url: " + folder_url
+    # message.attach_file('/templates/testPage.html')
+    if len(POSTMARK_TOKEN) == 0:
+        logger.info('Report not emailed, Postman token not set')
+    else:
+        send_email_postmark(subject, message, recipient_email)
+        logger.info('Email sent to ' + recipient_email)
+
+
+def send_email_postmark(subject, message, recipient_email):
```
Comment (SriMouli) [inline engine/email_generator.py:50]: change to support any generic smtp client

File: engine/selenium_code.py [hunk 10]
```diff
@@ -0,0 +1,425 @@
+"""
```
Comment (SriMouli) [inline engine/selenium_code.py:1]: rename to selenium_helper  or suitable name

**Review by SriMouli — CHANGES_REQUESTED** [review]
verify is [pep8](https://www.python.org/dev/peps/pep-0008/) is follwed

File: README.md (hunk 2 above)
Comment (rajatsharda04) [reply README.md:11]: done

File: engine/selenium_code.py (hunk 10 above)
Comment (rajatsharda04) [reply engine/selenium_code.py]: Done

File: docker-compose.yml (hunk 8 above)
Comment (rajatsharda04) [reply docker-compose.yml]: Done

File: daksha/settings.py (hunk 6 above)
Comment (rajatsharda04) [reply daksha/settings.py]: Done

File: README.md (hunk 3 above)
Comment (rajatsharda04) [reply README.md]: Done

File: daksha/settings.py (hunk 4 above)
Comment (rajatsharda04) [reply daksha/settings.py]: Done

File: docker-compose.yml (hunk 7 above)
Comment (rajatsharda04) [reply docker-compose.yml]: Done

File: daksha/settings.py (hunk 5 above)
Comment (rajatsharda04) [reply daksha/settings.py:44]: Done

File: Dockerfile (hunk 1 above)
Comment (rajatsharda04) [reply Dockerfile:15]: Added run CMD to docker file.

File: engine/email_generator.py (hunk 9 above)
Comment (rajatsharda04) [reply engine/email_generator.py:50]: Created issue https://github.com/mykaarma/Daksha/issues/2

File: docker-compose.yml [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    build: .
+    container_name: daksha
+    restart: always
+    volumes:
+      - /daksha/test-data:/daksha/test-data
+    ports:
+      - "8083:8000"
+    environment:
+      STORAGE_PATH: "/daksha/test-data"
+      APACHE_URL: ""
```
Comment (SriMouli) [inline docker-compose.yml:14]: add some sensible defaults where possible.

**Review by SriMouli — APPROVED** [review]

**Pull request merged**